### PR TITLE
Thunderbolt: Add support for TitanRidge Thunderbolt controllers

### PR DESCRIPTION
**Pull request by finikorg**
Add verification of TitanRidge Thunderbolt Controllers. Support building cab image from binary.

### CODE REVIEW

File: plugins/thunderbolt/fu-thunderbolt-image.c [hunk 1]
```diff
@@ -434,6 +456,9 @@ get_device_locations (guint16 id)
 	case 0x15D3:
 	case 0x15DA:
 	case 0x15C0:
+	case 0x15E7:
+	case 0x15EA:
+	case 0x15EF:
```
Comment (ybernat) [inline plugins/thunderbolt/fu-thunderbolt-image.c]: Oops, I forgot that this location isn't relevant for TR devices anymore. The "X of N" info is now part of the DROM. Let's discuss it.

**Comment by hughsie** [comment]
Looks okay to me, but I guess @ybernat needs the last word.

File: plugins/thunderbolt/fu-thunderbolt-image.c (hunk 1 above)
Comment (superm1) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: @ybernat so if just this part of PR is dropped is it OK then?  Or is this a more involved change?

Comment (ybernat) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: No, it needs to add a new function to validate the "X of N" info from the DROM for those devices.
If preferred, it's possible to split it to 2 PRs. First add only the TR host support from this PR (by removing the IDs from this function, so it returns `NULL` and fails the validation) and then add the TR device support on a later PR.

Comment (superm1) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: Yeah I'd like to see TR host in 1.0.6 if possible, so splitting the two makes sense to me. 
I think once @hughsie has the capsule cleanup PR done (and probably baking for a few days) he wants to cut 1.0.6.  If @finikorg can get the second PR for TR device done and tested before then, even better :)

File: plugins/thunderbolt/fu-thunderbolt-image.c [hunk 2]
```diff
@@ -434,6 +456,9 @@ get_device_locations (guint16 id)
...
 	case 0:
```
Comment (ybernat) [inline plugins/thunderbolt/fu-thunderbolt-image.c]: Does anyone remember by any chance why 0 is here? (*embarrassed icon*)

Comment (superm1) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: If I was gonna guess I'd say it's one of these:
1. it's for CRB/SDV
2. safe mode (let them use force)

File: plugins/thunderbolt/fu-thunderbolt-image.c (hunk 1 above)
Comment (finikorg) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: done

File: plugins/thunderbolt/fu-thunderbolt-image.c (hunk 2 above)
Comment (ybernat) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: Rereading the comments on the original PR, it seems like we did it to be future-proof for devices.
https://github.com/hughsie/fwupd/pull/201#pullrequestreview-58681545

Assuming that the comparison of vendor and model IDs are enough to be safe.

(safe-mode isn't relevant. On safe-mode no device is recognized by FW as there is no FW running... And for host we must compare the swap-lane table to be sure it's OK and also the kernel driver must be updated anyway for it to work so it's fine to update fwupd too for this.)

The point is that now the 0 must return an empty array. Oops.

Comment (ybernat) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: @superm1 Does my reasoning here make sense? If so, I'd like to fix it before 1.0.6. (Should be quick and simple, and thus safe enough even w/o testing it, I think)

Comment (ybernat) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: (Ah, yes, this is what it returns at the end, line 608. If `id == 0` it returns `UNKNOWN_DEVICE`. OK, preparing a PR.)

Comment (superm1) [reply plugins/thunderbolt/fu-thunderbolt-image.c]: Okay

**Pull request merged**